### ISSUE TITLE
Governor reverting to performance

### ISSUE THREAD

**Issue by rekeeszeng** [vagnum08/cpupower-gui#22]
I've been using this for quite some time. I use a script to set the governor to powersave upon startup. Most of the time, it works well, but the governor randomly reverts to performance.
I thought that's because of a high load. So I ran some stress tests, which, however, didn't change the governor.
The change seems to be random. Is that a bug?

**Comment by vagnum08** [comment]
Hi, 

I don't think this is a bug of cpupower-gui since it doesn't run as a daemon. You can only change the values by running the command.

The only way that I can think of is if you are using something like tlp. TLP runs as a daemon and changes the governor based on various rules like if the power plug is connected to the laptop switch the governor to performance.

Does your script runs only upon startup?

**Comment by rekeeszeng** [comment]
Thanks for the prompt reply.
Yes, I have a startup script to change the governor to powersave. I didn't notice the plug-in state's relationship with the governor change, 'cause most of the time, I have a power cord and the script does its job on startup.
I did some testing just now. Yes, you nailed it! When I connected the power cord, the governor changed to peformance. When I disconnected it, the governor changed to powersave. I don't know what's reponsible for the switch. I checked all the processes and services, and nothing had a name similar to TLP.
With the governor at performance, the CPU always runs at a speed close to its maximum. That makes little sense.
Anyway, I'll try and find what works like TLP and I'll be waiting for a new version of cpupower-gui that runs as a daemon if that's planned.
Thanks a lot.


------------------&nbsp;Original&nbsp;------------------
